### PR TITLE
DB <--> Code convergence

### PR DESCRIPTION
**Pull request by valentinyanakiev**
- Indexes may or may not be needed, but this would require a more thorough, case-by-case, performance analysis, with query plans etc
- aligned DB with code
- where space --> parentSpace referential integrity checks failed, set it to null. This may NOT be needed, if orphan data script fixes the data properly
- the following spaces on prod had issues (I've checked a few, they were not accessible):

```
  '2863d59e-deac-417d-85d3-7d02092d1000',
  '61bded0a-59a3-4bb2-a0b7-68ae9a3d1a6f',
  '6fd0eb2e-809d-4bdf-8d7c-b836fccafc24',
  '76b9bc76-fc97-48bd-918f-de133c68984c',
  '97973ada-2aa2-4ad1-a492-5a68f75eb9be',
  'aeba7a5a-910c-45b2-81de-7c2f08faa6c5',
  'd7e675e7-3c9c-42b2-ad4f-3650bf3635d9',
  'db7f587e-d1fa-423f-8923-88b5cd338be3',
  'e72e430d-0542-4878-983e-abf4e5fc2318',
  'ec7b85eb-def4-4c97-86a0-fb895fd8e8e8'
```

- I have NOT looked into fixing deeper, circular dependencies, that may require deeper architectural discussion. Raised a [story](https://app.zenhub.com/workspaces/alkemio-development-5ecb98b262ebd9f4aec4194c/issues/gh/alkem-io/server/4529) to follow that up, and put a comment in the code

For me, code was running fine after executing this, I've done exploratory testing

With those changes applied, running `npm run migration:generate XXX` doesn't generate delta.

## Summary by CodeRabbit

- **New Features**
	- Introduced a utility function for safely adding foreign key constraints, enhancing migration robustness.
	- Added a utility function for safely dropping indexes, ensuring integrity during database modifications.

- **Bug Fixes**
	- Improved handling of foreign key constraints in the `space` table to maintain data integrity.

- **Documentation**
	- Enhanced clarity in migration processes with additional comments and logging for foreign key and index operations.

- **Chores**
	- Reformatted code for improved readability and maintainability across migration files.

### CODE REVIEW

File: src/migrations/1726060322894-indexConvergance.ts
Comment (techsmyth) [inline src/migrations/1726060322894-indexConvergance.ts]: Should the name not match the file?

**Review by techsmyth — COMMENTED** [review]
Tried the following flow:
- dropped the db
- checked out develop
- ran migrations 
- checked out this brand
- ran migrations, got error:
image

Also raised a separate issue that some FKs do not start with FK_ , spotted via Index decorators. Maybe take that also here, or leave for later - your call.

File: src/migrations/1726060322894-indexConvergance.ts
Comment (valentinyanakiev) [reply src/migrations/1726060322894-indexConvergance.ts]: Addressed, nice catch

File: src/domain/timeline/event/event.entity.ts
Comment (valentinyanakiev) [reply src/domain/timeline/event/event.entity.ts:28]: Please investigate :)

**Comment by valentinyanakiev** [comment]
> Tried the following flow:
> 
> * dropped the db
> * checked out develop
> * ran migrations
> * checked out this brand
> * ran migrations, got error:
>   image
> 
> Also raised a separate issue that some FKs do not start with FK_ , spotted via Index decorators. Maybe take that also here, or leave for later - your call.

Fixed that with a lot of defensive code. Now the issue is, baseline migration diverges from prod DB. Given that `npm run migration:generate XXX` generates the delta between the DB in the connection string and the entity definitions. That delta is different between the prod base schema and a brand new DB created with migrations. So clearly, we have done something wrong in the past and we can't have 2 sources of truth easily, not sure what's the optimal course of action here.

The most efficient way may be to collapse the migration to one baseline again...

**Review by techsmyth — APPROVED** [review]
This brings convergence with production db, not with locally generated db. Agreed to reset migrations for development once this is merged / deployed.

**Pull request merged**
